diff -ru -x config.guess -x config.sub original/ncurses-5.9/form/fty_num.c archives/ncurses-5.9/form/fty_num.c
--- original/ncurses-5.9/form/fty_num.c	2010-01-24 06:14:36.000000000 +0900
+++ archives/ncurses-5.9/form/fty_num.c	2013-08-11 22:08:16.007597391 +0900
@@ -36,6 +36,7 @@
 
 MODULE_ID("$Id: fty_num.c,v 1.28 2010/01/23 21:14:36 tom Exp $")
 
+#undef HAVE_LOCALE_H
 #if HAVE_LOCALE_H
 #include <locale.h>
 #endif
